src/main/java/io/jenkins/plugins/opentelemetry/job/OtelTraceService.java

```diff
@@ -241,9 +241,11 @@ public Tracer getTracer() {
 
         @Override
         public String toString() {
+            // clone the Multimap to prevent a ConcurrentModificationException
+            // see https://github.com/jenkinsci/opentelemetry-plugin/issues/129
             return "RunSpans{" +
-                    "runPhasesSpans=" + runPhasesSpans +
-                    ", pipelineStepSpansByFlowNodeId=" + pipelineStepSpansByFlowNodeId +
+                    "runPhasesSpans=" + new ArrayList(runPhasesSpans) +
```

Comment:
This list will not need to modify, it is possible to use an unmodifiableList list

```suggestion
                    "runPhasesSpans=" + Collections.unmodifiableList(runPhasesSpans) +
```